File: Incubation/tags/EDSN_JUNE_2013/psim/src/org/space/oss/psim/spacecraft/SpacecraftFactory.java
package org.space.oss.psim.spacecraft;

import java.util.List;

import org.space.oss.psim.PSim;
import org.space.oss.psim.Spacecraft;

public interface SpacecraftFactory 
{
	List<Spacecraft> makeSpacecraft(PSim psim);
}

File: Incubation/branches/psim-scala-edsn/src/main/scala/org/space/oss/psim/spacecraft/SpacecraftFactory.scala
package org.space.oss.psim.spacecraft

import org.space.oss.psim.Spacecraft

trait SpacecraftFactory {
  def makeInstance(id:Int): Spacecraft
}

class DeafultSpacecraftFactory extends SpacecraftFactory {
  override def makeInstance(id:Int) = new SpacecraftBase("SC-"+id)
}